internal/rbd/rbd_util.go

```diff
@@ -902,58 +933,99 @@ func (rv *rbdVolume) getImageInfo(ctx context.Context, monitors string, cr *util
 	return imgInfo, nil
 }
 
-// snapInfo strongly typed JSON spec for snap ls rbd output
-type snapInfo struct {
-	ID        int64  `json:"id"`
-	Name      string `json:"name"`
-	Size      int64  `json:"size"`
-	Timestamp string `json:"timestamp"`
+func getCloneDepth(ctx context.Context, monitors, poolName, imageName string, depth int, cr *util.Credentials) (int, error) {
```

Comment:
can this be a method of rbdVolume instead? you're passing all important members of the type, and are constructing a new object from those

no need to pass `depth`?